lib/common/zstd_internal.h

```diff
@@ -196,6 +196,7 @@ static void ZSTD_copy8(void* dst, const void* src) { memcpy(dst, src, 8); }
 #define COPY8(d,s) { ZSTD_copy8(d,s); d+=8; s+=8; }
 static void ZSTD_copy16(void* dst, const void* src) { memcpy(dst, src, 16); }
 #define COPY16(d,s) { ZSTD_copy16(d,s); d+=16; s+=16; }
+#define COPY32(d,s) { COPY16(d,s); COPY16(d,s) }
 
 #define WILDCOPY_OVERLENGTH 32
 #define WILDCOPY_VECLEN 16
```

Comment:
If you want to copy 32 bytes at a time, you have to bump this to 32. This tells the decompressor how large the offset has to be for wildcopy to be used. If the offset is < 32 bytes, and you do a 32 byte memcpy, it is undefined behavior.

Reply:
Thanks